lib/puppet/parser/functions/compliance_map.rb

```diff
@@ -239,10 +248,15 @@ def self.lookup_global_silent(param)
         # Super-hack because 3.X raises a Puppet::ParseError complaining about
         # the lookup function
         rescue NoMethodError, Puppet::ParseError
+          reference_map = nil
+
           begin
-            reference_map = call_function('hiera_hash',['compliance_map',nil])
+            if Puppet.version.split('.').first.to_i > 3
+              reference_map = call_function('hiera_hash',['compliance_map',nil])
+            else
+            end
```

Comment:
I'm confused by this logic, if we are running on Puppet 3, we call hiera_hash() on 'compliance_map' from top? Is this a compatibility shim so it can be merged back into simp 5? If so we need to document that here and make a ticket to remove the shim in 7.x